pkg/Microsoft.ML.OnnxTransform/Microsoft.ML.OnnxTransform.nupkgproj

```diff
@@ -7,7 +7,7 @@
 
   <ItemGroup>
     <ProjectReference Include="../Microsoft.ML/Microsoft.ML.nupkgproj" />
-    <PackageReference Include="Microsoft.ML.OnnxRuntime.Gpu" Version="$(MicrosoftMLOnnxRuntimeGpuPackageVersion)"/>
+    <PackageReference Include="Microsoft.ML.OnnxRuntime" Version="$(MicrosoftMLOnnxRuntimePackageVersion)"/>
```

Comment:
Why aren't we using the GPU package anymore?

Reply:
The GPU package will be available on Nuget.Org.  In particular for Linux, CPU execution is blocked if the libraries are not available.
Using the CPU package will enable Linux testing as well, if an Ubuntu CI leg is available in the future.

---
In reply to: [251941810](https://github.com/dotnet/machinelearning/pull/2085#discussion_r251941810) [](ancestors = 251941810)

Reply:
I’m not understanding this. Will ML.NET use the GPU package or not? Why did we switch to the GPU package before, only to switch back off the GPU package now?

Reply:
The primary reason is that OnnxRuntime GPU package won't run on Linux  without the CUDA libraries are installed, even if a user wants to just run on CPU. To avoid confusion, we should go with a default of a CPU package, which will run cross-platform (Ubuntu at least) without requiring any extra software. And then, to use the GPU package, request users to install CUDA and the GPU OnnxRuntime package. The out-of-box package should run everywhere without extra software. When we add MacOS support, the runtimes will be available only in the CPU package (there's no GPU runtime for Mac), so long term using the CPU package seems like a better option.

Most other frameworks have 2 separate packages.

See the  MXNet GPU support instructions here --> [https://mxnet.incubator.apache.org/versions/master/install/windows_setup.html#install-with-gpus](https://mxnet.incubator.apache.org/versions/master/install/windows_setup.html#install-with-gpus). 
And Theano as well --> [http://deeplearning.net/software/theano/install_windows.html](http://deeplearning.net/software/theano/install_windows.html).



---
In reply to: [252102461](https://github.com/dotnet/machinelearning/pull/2085#discussion_r252102461) [](ancestors = 252102461)

Reply:
OK, this works for now. I agree it is best to have our "default" work in as many places as possible.

> documentation currently says to use the Microsoft.ML.OnnxRuntime.Gpu package if users want to run on GPU. Currently they'll need to rebuild MLNET from source to do this. 

We typically don't take this route in .NET libraries*. Instead we typically provide all options in binary form, i.e. add an optional OnnxTransform-Gpu separate package/transform. 

(*) Reasons for this:
1. It means the files are no longer signed by Microsoft.
2. It messes with dependencies, because now if someone wanted to ship a package that depended on the OnnxTransform.Gpu package, they would need to redistribute it themselves. And if 2 people wanted to do it, they would have separate packages.